src/components/PopupWithForm.js

```diff
@@ -7,6 +7,18 @@ class PopupWithForm extends Popup {
     this._form = this._popupElement.querySelector(".form");
     this._handleFormSubmit = handleFormSubmit;
     this._formInputs = this._popupElement.querySelectorAll(".form__input");
+
+    this._submitButton = this._popupElement.querySelector(".form__button");
+
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`